storage/ndb/src/kernel/vm/SimulatedBlock.cpp

```diff
@@ -6194,9 +6205,14 @@ Uint32 SimulatedBlock::get_lqhkeyreq_ref(DistributionHandler * const handle,
       block_num = DBQLQH;
     }
   }
+  if (rr_info->m_total_weight == 0)
```

Comment:
Line 6198 seems to have a problem:
```c++
query_instance_no = get_shared_ldm_instance(query_instance_no);
``` 
m_threads[][].m_shared_instance is initialized at thr_config.cpp:714, it seems like be overwritten again and again there.